### ISSUE TITLE
Ability to Clone/Copy a collection

### ISSUE THREAD

**Issue by simonhayward** [typesense/typesense#407]
## Description

We have an almost continual background indexing process which cycles through and creates a collection and once this process is complete we switch out the collection to use an alias and delete the previous collection.

We would benefit from being able to clone an existing collection in-place on the server and then remove/update parts of this index and then switch it out to use an alias and delete the previous collection.

This would speed up our indexing process significantly, rather than having to do a blanket update each time. 

We do not at the moment target individual records to be updated but do know based on groupings/sections those we want to update in greater frequency than other groupings/sections. If we could clone a collection and manage which sections within that index we delete/create would greatly reduce our indexing time as we want to increase the frequency and speed of the collection creation which we use to populate on a background process which is referenced via an alias.

## Metadata

**Typsense Version**: Typesense v0.21.0

**OS**: The typesense cloud based service.

**Comment by jasonbosco** [comment]
You can now clone a collection's schema, synonyms and overrides in v0.24: https://typesense.org/docs/0.24.0/api/collections.html#cloning-a-collection-schema

**Comment by brideo** [comment]
Is it possible to copy/move a collection with the documents? Or put a request for this feature in?

**Comment by jasonbosco** [comment]
It's not possible to clone the documents in a collection. Could you create a separate issue for it?
